code/modules/power/engines/supermatter/supermatter_event.dm

```diff
@@ -30,30 +30,26 @@
 	on_start()
 	alert_engi()
 	if(duration)
-		addtimer(CALLBACK(src, PROC_REF(on_end)), 30 SECONDS)
+		addtimer(CALLBACK(src, PROC_REF(on_end)), duration)
 
 /datum/supermatter_event/proc/on_start()
 	return
 
 /datum/supermatter_event/proc/alert_engi()
 	return
 
-/datum/supermatter_event/proc/after_duration()
-	// Normally, just call on end. Some s-class events require a 2 timers
-	on_end()
-
 /datum/supermatter_event/proc/on_end()
 	sm_radio_say("Anomalous crystal activity has ended.")
 	supermatter.heat_penalty_threshold = HEAT_PENALTY_THRESHOLD
 	supermatter.gas_multiplier = 1
 	supermatter.power_additive = 0
 	supermatter.event_active = null
-	supermatter.last_event = src
+	supermatter.last_event = world.time
```

Comment:
This was intended to be the ID of the last event, for display purposes. We likely need a time as well for this